docs/src/reference/api/cmfpackage.json/index.md

```diff
@@ -0,0 +1,199 @@
+>
+> #### ignoreFiles
+> File pattern that should point to `.cmfpackageignore` files.
+>
+> #### action
+> Action that will occur during the packing. Defined via an Enum, check all the valid values [here](packaction.md).
+
+The properties **source** and **target** have support for token replacement of any property of the cmfpackage.
+Example:
+```json
+{
+  "packageId": "Cmf.Custom.Database.Post",
+  "version": "1.1.0",
+  "description": "Cmf Custom Database Post Scripts Package",
+  "packageType": "Database",
+  "isInstallable": true,
+  "isUniqueInstall": false,
+  "contentToPack": [
+    {
+      "source": "Online/$(version)/*",
+      "target": "Online/$(version)"
+    }
+  ]
+}
+```
+This means that the contentToPack will look in to the folder `/Online/1.1.0`.
+
+
+### dependencies
+Dependencies are specified in an array of a simple object that maps a packageId to a version.
+It can point to a local dependency (in the same repo) or to a remote dependency (in a remote repo).
+> Remote dependencies depend on remote repos(currently we only support folders), these repos are defined in the file [`repositories.json`](../repositores.json)
+
+Example:
+```json
+{
+    "packageId": "Cmf.Custom.Data",
+    "version": "1.1.0",
+    "description": "Cmf Custom Data Package",
+    "packageType": "Data",
+    "isInstallable": true,
+    "isUniqueInstall": true,
+    "dependencies": [
+    {
+      "id": "Cmf.Custom.Business",
+      "version": "1.0.0"
+    }]
+}
+```
+
+
+### relatedPackages
+In some cases, you want to guarantee that a set packages are built or packed together.
+To do this you just need to add this property, point to the relativePath of the package and define when it should be build or packed.
+Example:
+```json
+{
+    "packageId": "Cmf.Custom.Data",
+    "version": "1.1.0",
+    "description": "Cmf Custom Data Package",
+    "packageType": "Data",
+    "isInstallable": true,
+    "isUniqueInstall": true,
+    "relatedPackages": [
+    {
+        "path": "../Cmf.Custom.Business",
+        "preBuild": true,
+        "postBuild": false,
+        "prePack": false,
+        "postPack": false
+    }]
+}
+```
+
+### testPackages
+Just like the `dependencies` property, the `tespackages` are specified in an array of a simple object that maps a packageId to a version.
+This is useful to link packages of type Test, to any other packages. This allow the [`cmf assemble`](../../commands/assemble.md) command to assemble it together with the relative package.
+
+## **Generic Type Packages**
+This type doesn't have any predefined BuildStep, Step or ContentToPack, so it will completely rely on what is defined to know how it should be built, packed and deployed. Check the above properties that are only available for this PackageType.
+
+### buildSteps
+Array of commands that will be used to build the package during the `cmf build` command execution.
```

Comment:
probably worth noting that these are not CLI internal tasks but terminal commands, similar to package.json scripts

Reply:
I will add an example